src/main/java/org/mule/extension/http/internal/listener/HttpRequestAttributesBuilder.java

```diff
@@ -44,9 +47,15 @@ public HttpRequestAttributes build() {
     String version = request.getProtocol().asString();
     String scheme = requestContext.getScheme();
     String method = request.getMethod();
-    String uri = request.getUri();
-    String path = extractPath(uri);
-    String queryString = extractQueryParams(uri);
+
+    URI uri = request.getUri();
+    String path = uri.getPath();
+    String uriString = path;
+    Optional<String> query = ofNullable(uri.getQuery());
```

Comment:
What's the point of generating an Optional just to check `isPresent` afterwards?